src/bun.js/webcore/response.zig

```diff
@@ -2049,6 +2116,59 @@ pub const Fetch = struct {
                             }
                         }
 
+                        if (options.get(ctx, "ssl_props")) |ssl_opts| {
+                            if (!ssl_opts.isEmptyOrUndefinedOrNull() and ssl_opts.isObject()) {
+                                if (ssl_opts.get(ctx, "pfx")) |pfx_opt| {
+                                    if (pfx_opt.isString() or pfx_opt.isBuffer(ctx.ptr())) {
+                                        const err = JSC.createError(globalThis, "Pfx is not yet supported", .{});
+                                        return JSPromise.rejectedPromiseValue(globalThis, err);
+                                    }
+
+                                    // TODO(Liz3); Actually implement this
+                                    // if (pfx_opt.isString())
+                                    //     pfx = pfx_opt.toSlice(ctx.ptr(), allocator).slice();
+                                    // if (pfx_opt.isBuffer(ctx.ptr())) {
+                                    //     if (JSC.Node.StringOrBuffer.fromJS(globalThis, allocator, pfx_opt, exception)) |sb| {
+                                    //         pfx = sb.slice();
+                                    //     }
+                                    // }
+                                }
+                                if (ssl_opts.get(ctx, "ca")) |ca_opt| {
+                                    if (ca_opt.isString())
+                                        ca = ca_opt.toSlice(ctx.ptr(), allocator).slice();
+
+                                    if (ca_opt.isBuffer(ctx.ptr())) {
+                                        if (JSC.Node.StringOrBuffer.fromJS(globalThis, allocator, ca_opt, exception)) |sb| {
+                                            ca = sb.slice();
+                                        }
+                                    }
+                                }
+                                if (ssl_opts.get(ctx, "cert")) |cert_opt| {
+                                    if (cert_opt.isString())
+                                        cert = cert_opt.toSlice(ctx.ptr(), allocator).slice();
```

Comment:
This leaks memory

Reply:
as it seams i falsely assumed the behaviour to be the same as C++ `std::string`, what would be the correct approach here?

Reply:
hmm also we sould use a null-terminated string like in `hostname`here right? `toOwnedSliceZ`